spring-core/src/main/java/org/springframework/core/annotation/AnnotationUtils.java

```diff
@@ -164,9 +164,6 @@ public static boolean isCandidateClass(Class<?> clazz, Class<? extends Annotatio
 	 * @see #isCandidateClass(Class, Class)
 	 */
 	public static boolean isCandidateClass(Class<?> clazz, String annotationName) {
-		if (annotationName.startsWith("java.")) {
-			return true;
-		}
```

Comment:
This check is not redundant.

This checks the annotation type; whereas, the subsequent `hasPlainJavaAnnotationsOnly()` invocation checks the candidate class.

Thus, please revert this change.